integration-tests/runtime-attach/runtime-attach-app/src/main/java/co/elastic/apm/testapp/AppMain.java

```diff
@@ -50,18 +54,36 @@ public static void main(String[] args) {
         }
 
         boolean selfAttach = false;
+
         if (args.length > 1) {
             selfAttach = args[1].equals("self-attach");
         }
 
+        File agentConfig = null;
+        if (args.length > 2) {
+            agentConfig = new File(args[2]);
+            if (!Files.isReadable(agentConfig.toPath())) {
+                throw new IllegalStateException("missing/invalid agent configuration " + agentConfig);
+            }
+        }
+
         registerJmx(app);
 
         if (selfAttach) {
-            System.out.println("Using self-attach");
+            if (agentConfig == null) {
+                System.out.format("Using self attach with classpath configuration\n");
+                ElasticApmAttacher.attach();
+            } else {
+                System.out.format("Using self attach with external configuration %s\n", agentConfig.getAbsolutePath());
+                Map<String, String> config = new HashMap<>();
+                config.put("service_name", "self-attach-external-config");
+                config.put("config_file", agentConfig.getAbsolutePath());
+                ElasticApmAttacher.attach(config);
+            }
             ElasticApmAttacher.attach();
```

Comment:
shouldn't this be removed (the attach happens in the `then` and `else` clauses?

Reply:
Good catch, trying to attach twice will be ignored in practice if already attached.